matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/homeserver/GetHomeServerCapabilitiesTask.kt

```diff
@@ -87,7 +87,7 @@ internal class DefaultGetHomeServerCapabilitiesTask @Inject constructor(
 
         val mediaConfig = runCatching {
             executeRequest(globalErrorReceiver) {
-                mediaAPI.getMediaConfig()
+                mediaAPIProvider.getMediaAPI().getMediaConfig()
```

Comment:
There is a problem here since you're calling `mediaAPIProvider.getMediaAPI()` before calling `capabilitiesAPI.getVersions()`, which will tell which media API can be used.